### PR TITLE
Flow tools, addins and modules paths from the botstrapper to cake.exe

### PR DESCRIPTION
**Pull request by paulomorgado**
This change allows any change to the tools, addins or modules paths in the bootstrapper to flow to the invocation of cake.exe

### CODE REVIEW

File: cake.config [hunk 1]
```diff
@@ -8,8 +8,8 @@ LoadDependencies=false
 
 [Paths]
 Tools=./tools
```
Comment (gep13) [inline cake.config]: @paulomorgado wouldn't be change also be required here?

File: packages.config [hunk 2]
```diff
@@ -1,4 +1,4 @@
 <?xml version="1.0" encoding="utf-8"?>
 <packages>
-    <package id="Cake" version="0.30.0" />
+    <package id="Cake" version="0.19.1" />
```
Comment (gep13) [inline packages.config]: @paulomorgado this change shouldn't be required.

**Comment by gep13** [comment]
@paulomorgado @masterzdran these suggested changes would have quite far reaching additional changes in terms of documentation, and other parts of Cake (i.e. extensions, etc), so this is not something that will be pulled in immediately, if at all.  We will need to discuss this within the team, alongside our other discussions about changes to bootstrappers, etc.

File: cake.config (hunk 1 above)
Comment (paulomorgado) [reply cake.config]: Fixed!

**Comment by gep13** [comment]
>@paulomorgado said...
I understand that, @gep13 but, do you really think tools is such a unique name that will unlikely colide with other artifacts?

In the 4+ years that Cake has been an active project, I am only aware of 1 other person raising the naming convention of the tools folder as an issue.  Once we provided the ability to configure the name of the tools folder, we haven't heard any other complaints/concerns until now.

**Comment by paulomorgado** [comment]
Fair enough, @gep13.

>If I had asked people what they wanted, they would have said faster horses.

_Henry Ford_

😄

**Comment by vpaulino** [comment]
Naming environment variables according to the scope of their usage is important because it reduces the probabillity of name collisions from different ( tooling/sdk/thirth party libs or frameworks ) and software in that same environment.  Now more then ever the usage of env variables is getting more relevant with software that runs on controlled environments such as docker containers... imagine now if another set of tools use the same env names that are not scoped? On the other end ... With the current namming convention Cake is only making sure that no other tool uses the name TOOLS ...wich is good   then no other tool will make the same mistake ! :)

**Comment by devlead** [comment]
@vpaulino true.

That said all Cake related [environment variables](https://cakebuild.net/docs/fundamentals/default-configuration-values) are prefixed `CAKE_` so whilst there's a chance of collision there, it's not super likely.

Tools folder is the current convention, it's a suggestion and your free to change as you wish for your project.

With millions of builds done with Cake, where in general a bit hesitant to change defaults, as that might disrupt for more people than it benefits.

We've just got 2 requests in 4.5 years, so don't think it's a wide spread major issue.

**Comment by pascalberger** [comment]
We did some maintenance work on this repo which included removing of the `develop` branch which closed this PR.

As mentioned in previous comments it is unlikely that we switch names of folder, especially since in the meantime we recommend [Cake .NET Tool](https://cakebuild.net/docs/running-builds/runners/dotnet-core-tool). Feel free to [start a new discussion](https://github.com/cake-build/cake/discussions/category_choices) if this is still an issue.

**Pull request closed without merging**